
\documentclass[12pt,letterpaper]{article}

%%%%% packages provided by sys bio template
\usepackage{fixltx2e}
\usepackage{textcomp}
\usepackage{fullpage}
\usepackage{amsfonts}
\usepackage{verbatim}
%% \usepackage[english]{babel} %% I use polyglossia instead
%% not needed: \usepackage{pifont}
\usepackage{color}
\usepackage{setspace}
\usepackage{lscape}
\usepackage{indentfirst}
\usepackage[normalem]{ulem}
\usepackage{booktabs}
%\usepackage{nag}
\usepackage{natbib}
%\usepackage{bibtex}
\usepackage{float}
\usepackage{latexsym}
%\usepackage{hyperref}
\usepackage{url}
%\usepackage{html}
\usepackage{hyperref}
\usepackage{epsfig}
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{bm}
\usepackage{array}
%\usepackage{mhchem}
\usepackage{ifthen}
%% I am already using it: \usepackage{caption}
\usepackage{hyperref}
%\usepackage{xcolor}
\usepackage{amsthm}
\usepackage{amstext}
%%%%


% line numbers
\usepackage{lineno}
\modulolinenumbers[5]
\linenumbers

\usepackage{natbib}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont[Mapping=tex-text]{Crimson Text}
\setsansfont{SourceSansPro-Regular}

%%% syst bio options
\linespread{1.66}
% All text should be double-spaced
% with occasional exceptions for tables.
\raggedright
\setlength{\parindent}{0.5in}

\setcounter{secnumdepth}{0}

% Our sections are not numbered and our papers do not have
% Tables of Contents. We don't
% present a list of figures or list of tables, either.

% Any common font is fine.
% (A common sans-serif font should be used on figures, but figures should be
% separate from the LaTeX document.)

\pagestyle{empty}

\renewcommand{\section}[1]{%
\bigskip
\begin{center}
\begin{Large}
\normalfont\scshape #1
\medskip
\end{Large}
\end{center}}

\renewcommand{\subsection}[1]{%
\bigskip
\begin{center}
\begin{large}
\normalfont\itshape #1
\end{large}
\end{center}}

\renewcommand{\subsubsection}[1]{%
\vspace{2ex}
\noindent
\textit{#1.}---}

\renewcommand{\tableofcontents}{}

\bibpunct{(}{)}{;}{a}{}{,}  % this is a citation format command for natbib
%%% end syst bio options


\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

% ----------------------------------------------------------------------------- %

\begin{document}

\begin{flushright}
Version dated: \today
\end{flushright}

\bigskip
\noindent AN R PACKAGE TO INTERACT WITH THE OPEN TREE OF LIFE

\bigskip
\medskip

\begin{center}
  \noindent{\Large \bf \texttt{rotl} an R package to interact with the Open Tree
    of Life Data}

\bigskip

\noindent{\normalsize \sc Fran\c{c}ois Michonneau}\\
\noindent {\small \it $^1$Florida Museum of Natural History \& Whitney Laboratory
  for Marine Sciences, University of Florida, Gainesville,
  FL 32611-7800, USA}\\
\end{center}
\medskip

\noindent{\bf Corresponding author:} Fran\c{c}ois Michonneau, Division of
Invertebrate Zoology, Florida Museum of Natural History, Gainesville, FL
32611-7800, USA; E-mail: francois.michonneau@gmail.com\\


\vspace{1in}

\subsubsection{Abstract}

Abstract goes here.


\vspace{1.5in}

Generating phylogenies has been greatly facilitated by advances in sequencing
and computing technologies. In parallel, new phylogenetic comparative methods
have been developed and make use of these phylogenies to explore fundamental
questions about the origin of biodiversity including the evolution of
morphological and ecological traits, the spatio-temporal variation in speciation
rates, or both \citep{Pennell2013}. The development of these methods has
contributed to the need for a greater availability of high-quality phylogenetic
information across life. However, phylogenetic information is scattered, often
only available as image files within publications, and the variety of file
formats used to store phylogenetic data makes it difficult for researchers to
access, synthesize, and integrate this information into their own research
(\citealt{Stoltzfus2012}, but see \citealt{Cranston2014} for suggestions of best
practices).

The Open Tree of Life (OTL) project aims at assembling and synthesizing our
current understanding of phylogenetic relationships across all organisms on
Earth \citep{Hinchliff2015}, while providing tools and services that facilitate
access to this information. OTL combines taxonomic information that serves as
the backbone for the phylogenetic relationships, and published phylogenies to
elucidate relationships among taxa. This combination of information is used to
form the synthetic tree. Studies can be contributed to the synthetic tree
through the curator interface \url{https://tree.opentreeoflife.org/curator},
allowing the synthetic tree to be continuously updated as relationships are
elucidated or reevaluated. The current draft of the Open Tree of Life contains
2.3 million tips. Beyond obvious applications across the life sciences to
explore research questions in evolution, biodiversity, and conservation, the
resources OTL provides could be used for education and outreach (e.g.,
illustrating course material, develop outreach activities to explore
relationships among species).

% Because of its comprehensive nature, researchers have access to phylogenetic
% information for species that have yet to be included in any phylogenetic tree
% as their position is inferred based on taxonomy.

The Open Tree of Life project provides a web interface to explore, query and
download parts of the synthetic Open Tree
(\url{http://tree.opentreeoflife.org}). Additionally, the project provides an
Application Programming Interface (API) to the different services to work
programmatically with the underlying data. Here, we present \texttt{rotl} an R
package to interact with OTL's API using R syntax to search for, and import
phylogenetic trees and taxonomic information into R. Having access to this data
directly in R allows researchers to take advantage of the multiple packages that
are available to visualize, analyze and manipulate phylogenetic and comparative
data (ape, phytools, geiger, ggtree, phylobase, RNeXML, etc... cite task view
\url{https://cran.r-project.org/web/views/Phylogenetics.html}).


\section{Services provided by OTL}

The OTL project provides four resources that serve data to users through the
API:

\begin{enumerate}
\item The \emph{taxonomy} used as the backbone from the tree, the Open Tree
  Taxonomy;
\item The \emph{studies} and their associated trees some of which are chosen by
  curators to assemble the synthetic tree;
\item A \emph{taxonomic name resolution service} (TNRS) used to match taxon
  names to the Open Tree Taxonomy identifiers;
\item The \emph{synthetic tree} itself (the Open Tree).
\end{enumerate}

Additionally, the project provides access to other data, the Graph of Life, that
stores information about the underlying tree alignment graphs \cite{Smith2013}
used to synthesize the various sources of information from which the Open Tree
is assembled from. Data from this API endpoint is not intended for normal users
and its use will not be demonstrated in this paper but \texttt{rotl} has
functions to retrieve data it provides.

\texttt{rotl} gives users access to all endpoints provided by version 2 of the
OTL's APIs. Phylogenetic trees served by the API can be imported directly into
R's memory and are represented using the \texttt{ape} \citep{Paradis2004} tree
structure (objects of class \texttt{phylo}), or can be written to files in the
Newick, NEXUS \citep{Maddison1997} or NeXML \citep{Vos2012} file formats. This
allows researchers to use these trees either directly with other R packages, or
to be imported in other programs that can read in phylogenetic tree files.

The synthetic tree currently does not have any branch lengths associated with
it, therefore parametric comparative methods cannot be used directly on the
trees returned by the API. However, resources and methods are being developed to
add branch lengths information to these topological trees
\citep[e.g.,][]{Ksepka2015} or use topological trees to identify
phylogenetically equivalent species to increase overlap between chronograms and
species trait data \citep{Pennell2015}. Without branch lengths, these trees
remain useful to illustrate relationships among species, or to map a trait on a
phylogeny for instance.

%% any examples available?

% Tree versionning? Replicability of analyses

\section{Technical information about \texttt{rotl}}

Phylogenetic information retrieved from OTL is converted into
\texttt{ape::phylo} objects by \texttt{rotl} using the NEXUS Class Library
\citep[NCL,][]{Lewis2003} as implemented in the \texttt{rncl} package
\url{https://cran.r-project.org/package=rncl}. Using NCL provides robust and
efficient parsing of large trees that may contain singletons nodes labeled with
taxonomic information.

\section{Demonstrations}
\label{sec:demonstrations}

\subsection{Getting relationships from a list of taxa}
\label{sec:get-relationships}

To get the relationships among a set of taxa from the Open Tree, the taxa first
need to be matched against the Open Taxonomy using the TNRS. This step retrieves
the taxa identifiers that will be used to extract the relationships for the set
of requested taxa.

To illustrate how to obtain relationships from a set of taxa, we will rely on
OTL to draw a phylogenetic tree for a set of model organisms.

<<init, echo=FALSE>>=
library(rotl)
@

First, we use the function \texttt{tnrs\_match\_names} to match the taxon names to
their Open Taxonomy identifiers.

<<get_taxa, cache=TRUE>>=
taxa <- tnrs_match_names(names = c("Escherichia colli",
                                   "Chlamydomonas reinhardtii",
                                   "Drosophila melanogaster",
                                   "Arabidopsis thaliana",
                                   "Rattus norvegicus",
                                   "Mus musculus",
                                   "Cavia porcellus",
                                   "Xenopus laevis",
                                   "Saccharomyces cervisae",
                                   "Danio rerio"))
@

The function \texttt{tnrs\_match\_names} returns a data frame that lists the
Open Tree identifiers as well as other information to help users to ensure that
the taxa matched are the correct ones. Here, there is no ambiguity in the taxa
matched, however, as the Open Tree Taxonomy includes taxa from bacteria, plants,
and animals that are regulated by different nomenclatural codes (ICNP, ICN and
ICZN respectively), OTL and \texttt{rotl} provide tools to deal with potential
hemihomonyms. The argument \texttt{context\_name} can be used to limit potential
matches to a taxonomic group (see the function \texttt{tnrs\_contexts} for a list
of possible options).  When this strategy cannot be used in case the tree
encompasses multiple domains (as in the present example), the function
\texttt{inspect} lists alternative matches for a taxon name, and \texttt{update}
replaces it in the results. An example of this approach is provided in the
vignette ``How to use rotl?'' that accompanies the package.

By default, approximate matching is enabled when attempting to match taxonomic
names to their Open Tree Taxonomy identifiers. Additionally, taxonomic synonyms
are included in the Open Tree Taxonomy allowing researchers to match correct
identifiers for taxon names that might include misspellings or synonyms. These
features will facilitate the tedious data cleaning process often needed when
matching taxon names. In the example provided, both \textit{Escherichia coli}
and \textit{Saccharomyces cerevisiae} are misspelled but OTL's TNRS finds the
correct match for these taxa.

Now that the taxon names are matched to the Open Tree identifiers, we can pass
them to the function \texttt{tol\_induced\_subtree} to retrieve the
relationships among these taxa. In turn, the tree can be plotted directly as it
is returned as an \texttt{ape::phylo} object (Figure~\ref{fig:plot_taxa}).

<<plot_taxa, cache=TRUE, fig.cap="The phylogenetic tree returned by OTL for the list of model species used as an example.">>=
tree <- tol_induced_subtree(ott_ids = taxa[["ott_id"]])
plot(tree)
@

\subsection{Getting trees from studies}
\label{sec:get-tree-study}

\texttt{rotl} can also be used to retrieve trees accompanying studies that have
been submitted through the curator interface, and identify the trees that
contribute to the synthetic tree. These trees provide a useful resource to
reproduce or expand on an already published analysis.

Criteria that can be used to search for studies or their associated trees are
available through the output of the function
\texttt{studies\_properties}. Typically, users will want to search for studies
or trees based on taxon names (or their Open Taxonomy identifiers), but other
criteria such as the title of the publication can be used. Here we demonstrate
how to look for and retrieve a tree for study focusing on the family Felidae
(Figure~\ref{fig:plot_cats}).

<<get_cats, fig.cap="Tree from \\cite{} obtained from OTL">>=
cat_studies <- studies_find_studies(property = "ot:focalCladeOTTTaxonName",
                                    value = "felidae")
cat_studies
@

Currently only one study focused on this family is available from OTL, and a
single tree is associated with it. We can then retrieve the study and tree
identifiers, and pass them to the function \texttt{get\_study\_tree} to have the
tree in memory:

<<get_cat_tree, results='hide'>>=
cat_tree <- get_study_tree(study_id = cat_studies[["study_ids"]][1],
                           tree_id = cat_studies[["tree_ids"]][1])
cat_tree
@

<<print_cat_tree, echo=FALSE>>=
print(cat_tree, printlen = 4)
@

When more than one tree is available for a given study, the function
\texttt{list\_trees} returns a list containing the tree identifiers for each
study easily. Alternatively, the function \texttt{get\_study} returns all the
trees (by default as \texttt{ape::phylo} objects) associated with a particular
study. Additional details about the study (i.e., its metadata) can be obtained
using the function \texttt{get\_study\_meta}.


\subsection{How does \texttt{rotl} fit into the R package ecosystem?}
\label{sec:how-does-rotl-fit}

In recent years, R has become part of the toolbox of many researchers in
evolutionary biology. As many methods for comparative analyses are implemented,
and because it is a relatively easy to use programming language, R greatly
facilitates the analysis of large datasets, and allows researchers to combine
methods in novel ways. Additionally, as more data is made available online and
accessible using APIs, many R packages have been developed to take advantage of
these datasets, especially spearheaded by the organization rOpenSci
\url{https://ropensci.org}. \texttt{rotl} contributes to these initiatives and
data from OTL can be combined with other sources easily. For instance in
Figure~\ref{cat_map}, we show how we can obtain a map of the occurrences for
some of the cat species (genus \textit{Lynx}) included in the phylogeny above
using records for these species found in GBIF (code is available in
supplementary material).

<<cat_map, echo=FALSE, results='hide', message=FALSE, cache=TRUE, fig.cap='GBIF records for the species in \\textit{Lynx} included in the phylogeny associated with the study by \\citealt{Johnson2006}'>>=

suppressPackageStartupMessages(library(rgbif))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(maps))

cat_species <- cat_tree$tip.label
lynx_species <- grep("^Lynx", cat_tree$tip.label, value = TRUE)

gbif_keys <- sapply(lynx_species,
                    function(x) name_backbone(name = x)$speciesKey,
                    USE.NAMES = FALSE)

lynx_loc <- occ_search(taxonKey = gbif_keys, limit = 500,
                       return = "data", fields = "minimal",
                       hasCoordinate = TRUE)

lynx_loc <- do.call("rbind", lynx_loc)
lynx_loc[["decimalLatitude"]] <- as.numeric(lynx_loc[["decimalLatitude"]])
lynx_loc[["decimalLongitude"]] <- as.numeric(lynx_loc[["decimalLongitude"]])
lynx_loc <- lynx_loc[complete.cases(lynx_loc), ]

names(lynx_loc)[1] <- "Species"

world <- map_data("world")

ggplot(lynx_loc) +
    annotation_map(world, fill="gray40", color="gray40") +
    geom_point(aes(y = decimalLatitude, x = decimalLongitude, color = Species),
               alpha = 0.3, size = 2) +
    coord_map(projection = "mercator", orientation = c(90, 0, 0)) +
    xlab("Longitude") + ylab("Latitude")

@



\section{Conclusion}
\label{sec:conclusion}

Resource as useful as the data that is in it. To make data available, it needs
to follow conventions so it can be incorporated into the OTL, i.e., submit tree
files (not images) to public repositories.

OTL provides an easy-to-use interface to submit studies.

\section{Availability}
\label{sec:availability}

\texttt{rotl} is available both from the CRAN repository
(https://cran.r-project.org/package=rotl) and from GitHub
(https://github.com/ropensci/rotl).


\section{Acknowledgments}
\label{sec:acknowledgments}

* OpenTree of Life hackathon
* Scott Chamberlain for code review.

\bibliographystyle{sysbio}

\bibliography{rotl-manuscript}

\end{document}
