
\documentclass[12pt,letterpaper]{article}

%%%%% packages provided by sys bio template
\usepackage{fixltx2e}
\usepackage{textcomp}
\usepackage{fullpage}
\usepackage{amsfonts}
\usepackage{verbatim}
%% \usepackage[english]{babel} %% I use polyglossia instead
%% not needed: \usepackage{pifont}
\usepackage{color}
\usepackage{setspace}
\usepackage{lscape}
\usepackage{indentfirst}
\usepackage[normalem]{ulem}
\usepackage{booktabs}
%\usepackage{nag}
\usepackage{natbib}
%\usepackage{bibtex}
\usepackage{float}
\usepackage{latexsym}
%\usepackage{hyperref}
\usepackage{url}
%\usepackage{html}
\usepackage{hyperref}
\usepackage{epsfig}
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{bm}
\usepackage{array}
%\usepackage{mhchem}
\usepackage{ifthen}
%% I am already using it: \usepackage{caption}
\usepackage{hyperref}
%\usepackage{xcolor}
\usepackage{amsthm}
\usepackage{amstext}
%%%%

\usepackage{natbib}
\usepackage{lineno}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont[Mapping=tex-text]{Crimson Text}
\setsansfont{SourceSansPro-Regular}

%%% syst bio options
\linespread{1.66}
% All text should be double-spaced
% with occasional exceptions for tables.
\raggedright
\setlength{\parindent}{0.5in}

\setcounter{secnumdepth}{0}

% Our sections are not numbered and our papers do not have
% Tables of Contents. We don't
% present a list of figures or list of tables, either.

% Any common font is fine.
% (A common sans-serif font should be used on figures, but figures should be
% separate from the LaTeX document.)

\pagestyle{empty}

\renewcommand{\section}[1]{%
\bigskip
\begin{center}
\begin{Large}
\normalfont\scshape #1
\medskip
\end{Large}
\end{center}}

\renewcommand{\subsection}[1]{%
\bigskip
\begin{center}
\begin{large}
\normalfont\itshape #1
\end{large}
\end{center}}

\renewcommand{\subsubsection}[1]{%
\vspace{2ex}
\noindent
\textit{#1.}---}

\renewcommand{\tableofcontents}{}

\bibpunct{(}{)}{;}{a}{}{,}  % this is a citation format command for natbib
%%% end syst bio options


\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

% ----------------------------------------------------------------------------- %

\begin{document}

\begin{flushright}
Version dated: \today
\end{flushright}

\bigskip
\noindent AN R PACKAGE TO INTERACT WITH THE OPEN TREE OF LIFE

\bigskip
\medskip

\begin{center}
  \noindent{\Large \bf \texttt{rotl} an R package to interact with the Open Tree
    of Life Data}

\bigskip

\noindent{\normalsize \sc Fran\c{c}ois Michonneau}\\
\noindent {\small \it $^1$Florida Museum of Natural History \& Whitney Laboratory
  for Marine Sciences, University of Florida, Gainesville,
  FL 32611-7800, USA}\\
\end{center}
\medskip

\noindent{\bf Corresponding author:} Fran\c{c}ois Michonneau, Division of
Invertebrate Zoology, Florida Museum of Natural History, Gainesville, FL
32611-7800, USA; E-mail: francois.michonneau@gmail.com\\


\vspace{1in}

\subsubsection{Abstract}

Abstract goes here.


\vspace{1.5in}

Generating phylogenies has been facilitated by advances in sequencing and
computing technologies enabling the investigation of evolutionary relationships
among taxa. In parallel, new phylogenetic comparative methods have been
developed and use these phylogenies to explore fundamental questions about the
origin of biodiversity including the evolution of morphological and ecological
traits, the spatio-temporal variation in speciation rates, or both
\cite{Pennell2013}. The development of these methods has contributed to the need
for a greater availability of high-quality phylogenetic information across
life. However, phylogenetic information is scattered, often only available as
image files within publications, and the variety of file formats used to store
phylogenetic information makes it difficult for researchers to access and
integrate into their own research (\cite{Stoltzfus2012}, but see
\cite{Cranston2014} for suggestions of best practices).

The Open Tree of Life (OTL) project aims at assembling and synthesizing our
current understanding of phylogenetic relationships across all organisms on
Earth \cite{Hinchliff2015}, while providing tools and services that facilitate
access to this information. OTL combines taxonomic information that serves as
the backbone for the phylogenetic relationships, and published phylogenies to
elucidate relationships among taxa. This combination of information is used to
form the synthetic tree. Studies can be contributed to the synthetic tree
through the curator interface \url{https://tree.opentreeoflife.org/curator},
allowing the tree to be updated continuously. The current draft of the Open Tree
of Life contains 2.3 million tips. Beyond obvious applications across the life
sciences to explore research questions in evolution, biodiversity, and
conservation, the resources OTL provides could be used in education and outreach
(e.g., illustrating course material, develop outreach activities to explore
relationships among species).

% Because of its comprehensive nature, researchers have access to phylogenetic
% information for species that have yet to be included in any phylogenetic tree
% as their position is inferred based on taxonomy.

The Open Tree of Life project provides a web interface to explore, query and
download parts of the synthetic Open Tree
(\url{http://tree.opentreeoflife.org}). Additionally, the project provides an
Application Programming Interface (API) to the different services it offers
allowing users to work programmatically with the underlying data. Here, we
present \texttt{rotl} an R package that allows users to interact with the OTL's
API to search for and import phylogenetic trees and taxonomic
information. Having access to this data directly from within R allows
researchers to take advantage of the multiple R packages that are available to
visualize, analyze and manipulate phylogenetic trees (ape, phytools, geiger,
ggtree, phylobase, etc... cite task view).


\section{Presentation of the services provided by OTL}

Presentation of the OTL: synthesis, 2.3M tips, continuously updated. There are
therefore 4 resources that the OTL provides: (1) the taxonomic information used
as the backbone from the tree, (2) the trees and associated published studies
that are used to build the synthetic tree, (3) a taxonomic name resolution
service used to map species names to the Open Tree Taxonomy, (4) the synthetic
tree. Also, Graph of Life endpoint but not of use for normal users and beyond
the scope of this paper.

\texttt{rotl} provides access to these services so the data stored in OTL
(including trees) can be imported into R (either in memory or by writing to
files) to interact with other packages and be used in comparative analyses, or
they can be written to files so they can used by other programs.

Interface with both the synthetic tree from Open Tree of Life and particular
trees from studies. Open Tree also features taxonomy lookup that can be used
with partial matching.

The OTL's API, and in turn \texttt{rotl}, provides services (1) to download a
tree from a set of species names, this can be useful to illustrate a manuscript,
map the evolution of a character in a taxonomic group, etc....; (2) to download
a tree published in a paper that might be used to build the synthetic tree, this
can be useful to reproduce or expand on an already published analysis; (3) to
download information about the taxonomy across all organisms considered by OTL
in a phylogenetic context.


Tree versionning? Replicability of analyses


Limitations

No branch lengths for trees from the synthetic TOL, limit applications for
comparative data, but could be possible to use calibrations from time tree
databases. Or combined with partial chronograms and use \cite{phyndr}.

\section{Demonstrations}
\label{sec:demonstrations}

\subsection{Getting relationships from a list of taxa}
\label{sec:get-relationships}

\subsection{Getting tree from a given study}
\label{sec:get-tree-study}

* Which studies are available in OTL?
* How to look for them?

\subsection{Exploring the taxonomy}
\label{sec:exploring-taxonomy}


\subsection{How does rotl fit into the R package ecosystem?}
\label{sec:how-does-rotl-fit}



\section{Conclusion}
\label{sec:conclusion}

Resource as useful as the data that is in it. To make data available, it needs
to follow conventions so it can be incorporated into the OTL, i.e., submit tree
files (not images) to public repositories.

OTL provides an easy-to-use interface to submit studies.

\section{Availability}
\label{sec:availability}

\texttt{rotl} is available both from the CRAN repository
(https://cran.r-project.org/package=rotl) and from GitHub
(https://github.com/ropensci/rotl).


\section{Acknowledgments}
\label{sec:acknowledgments}

* OpenTree of Life hackathon
* Scott Chamberlain for code review.

\end{document}
