
\documentclass[12pt,letterpaper]{article}

%%%%% packages provided by sys bio template
\usepackage{fixltx2e}
\usepackage{textcomp}
\usepackage{fullpage}
\usepackage{amsfonts}
\usepackage{verbatim}
%% \usepackage[english]{babel} %% I use polyglossia instead
%% not needed: \usepackage{pifont}
\usepackage{color}
\usepackage{setspace}
\usepackage{lscape}
\usepackage{indentfirst}
\usepackage[normalem]{ulem}
\usepackage{booktabs}
%\usepackage{nag}
\usepackage{natbib}
%\usepackage{bibtex}
\usepackage{float}
\usepackage{latexsym}
%\usepackage{hyperref}
\usepackage{url}
%\usepackage{html}
\usepackage{hyperref}
\usepackage{epsfig}
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{bm}
\usepackage{array}
%\usepackage{mhchem}
\usepackage{ifthen}
%% I am already using it: \usepackage{caption}
\usepackage{hyperref}
%\usepackage{xcolor}
\usepackage{amsthm}
\usepackage{amstext}
%%%%

%% Highlighting
\usepackage{soul}


% line numbers
\usepackage{lineno}
\modulolinenumbers[5]
\linenumbers

\usepackage{natbib}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont[Mapping=tex-text]{Crimson Text}
\setsansfont{SourceSansPro-Regular}

%%% syst bio options
\linespread{1.66}
% All text should be double-spaced
% with occasional exceptions for tables.
\raggedright
\setlength{\parindent}{0.5in}

\setcounter{secnumdepth}{0}

% Our sections are not numbered and our papers do not have
% Tables of Contents. We don't
% present a list of figures or list of tables, either.

% Any common font is fine.
% (A common sans-serif font should be used on figures, but figures should be
% separate from the LaTeX document.)

\pagestyle{empty}

\renewcommand{\section}[1]{%
\bigskip
\begin{center}
\begin{Large}
\normalfont\scshape #1
\medskip
\end{Large}
\end{center}}

\renewcommand{\subsection}[1]{%
\bigskip
\begin{center}
\begin{large}
\normalfont\itshape #1
\end{large}
\end{center}}

\renewcommand{\subsubsection}[1]{%
\vspace{2ex}
\noindent
\textit{#1.}---}

\renewcommand{\tableofcontents}{}

\bibpunct{(}{)}{;}{a}{}{,}  % this is a citation format command for natbib
%%% end syst bio options


\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

% ----------------------------------------------------------------------------- %

\begin{document}

\begin{flushright}
Version dated: \today
\end{flushright}

\bigskip
\noindent AN R PACKAGE TO INTERACT WITH THE OPEN TREE OF LIFE

\bigskip
\medskip

\begin{center}
  \noindent{\Large \bf \texttt{rotl} an R package to interact with the Open Tree
    of Life Data}

\bigskip

\noindent{\normalsize \sc Fran\c{c}ois Michonneau$^{1,2}$, Joseph Brown$^3$, David Winter$^4$}\\

\noindent {\small \it $^1$ Whitney Laboratory for Marine Sciences, University of
  Florida, St. Augustine, FL, USA}\\
\noindent {\small \it $^2$Florida Museum of Natural History, University of
  Florida, Gainesville, FL, USA} \\
\noindent{\small \it $^3$University of Michigan, Ann Arbor, MI, USA} \\
\noindent{\small \it $^4$Arizona State University, Tempe, AZ, USA} \\
\end{center}
\medskip

\noindent{\bf Corresponding author:} Fran\c{c}ois Michonneau, Division of
Invertebrate Zoology, Florida Museum of Natural History, Gainesville, FL
32611-7800, USA; E-mail: francois.michonneau@gmail.com\\




\vspace{1in}

\subsubsection{Abstract}

While phylogenies have been getting easier to build, it has been difficult to
re-use, combine, and synthesize the information they provide because published
trees are often only available as image files, and taxonomic information is not
standardized across studies. The Open Tree of Life (OTL) project addresses these
issues by providing a digital tree that encompasses all organisms, built by
combining taxonomic information and published phylogenies. The project also
provides tools and services to query and download parts of this synthetic tree,
as well as the source data used to build it. Here, we present \texttt{rotl}, an
R package to search and download data from the Open Tree of Life directly in
R. \texttt{rotl} uses common data structures allowing researchers to take
advantage of the rich set of tools and methods that are available in R to
manipulate, analyze, and visualize phylogenies.


\vspace{1.5in}



Advances in sequencing and computing technologies have lead to a revolution in
systematic biology. The ability to routinely generate molecular datasets from
any extant organism has allowed researchers to resolve long-standing taxonomic
disputes and estimate phylogenies for previously understudied groups. In
parallel, the ease with which phylogenies can be estimated has spurred the
development of new phylogenetic comparative methods. These methods allow
researchers to explore fundamental questions about the origin of biodiversity
including the evolution of morphological and ecological traits, the
spatio-temporal variation in speciation rates, or both
\citep{OMeara2012, Pennell2013}.

Ideally, the ever increasing number of published phylogenies would contribute to
a synthesis of phylogenetic knowledge, ultimately leading to a better
understanding of the history of life while at the same time providing
high-quality phylogenetic information for use in comparative analyses. However,
in practice, synthesizing phylogenetic data is a difficult task. Phylogenetic
information is largely scattered, often only available as image files within
publications, and the lack of standardization to store and represent
phylogenetic data makes it difficult for researchers to access, synthesize, and
integrate this information into their own research (\citealt{Stoltzfus2012};
\citealt{Drew2013}; but see \citealt{Cranston2014} for suggestions of best
practices).

The Open Tree of Life (OTL) project aims at assembling and synthesizing our
current understanding of phylogenetic relationships across all organisms on
Earth while providing tools and services that facilitate access to this
information \citep{Hinchliff2015}. OTL combines taxonomic information that
serves as the backbone for the phylogenetic relationships, and published
phylogenies to elucidate relationships among taxa. This combination of
information is used to structure the comprehensive synthetic tree. Studies can
be contributed to the synthetic tree through a curator interface
(\url{https://tree.opentreeoflife.org/curator}), allowing the synthetic tree to
be continuously updated as relationships are elucidated or reevaluated. The
current draft of the OTL synthetic tree contains 2.3 million tips. Beyond
obvious applications across the life sciences to explore questions in evolution,
biodiversity, and conservation, the resources OTL provides are useful for
education and outreach (e.g., illustrating course material, or developing
outreach activities to explore relationships among species).

% Because of its comprehensive nature, researchers have access to phylogenetic
% information for species that have yet to be included in any phylogenetic tree
% as their position is inferred based on taxonomy.


The R programming language is a popular tool for phylogenetics and comparative
analysis. The R packages \texttt{ape} \citep{Paradis2004}, \texttt{phylobase}
\citep{phylobase-0.8.0}, and \texttt{RNeXML} \citep{Boettiger2015} each provide
functions to import and manipulate trees within R and save the results in
standard data formats. Additional packages including \texttt{phytools}
\citep{Revell2012}, \texttt{geiger} \citep{Pennell2014-geiger}, and
\texttt{ggtree} \citep{Yu-ggtree}, allow users to analyze and visualize data in
a phylogenetic context (see
\url{https://cran.r-project.org/web/views/Phylogenetics.html} for a
comprehensive list of phylogenetics packages in R). In addition to packages for
phylogenetic and comparative analyses, a growing number of R packages allows
users to query and access data from the web (e.g. \texttt{rFISHBASE}
\citep{Boettiger2012}, \texttt{rAvis} \citep{Varela2014}, and \texttt{paleoDB}
\citep{Varela2015}), such that data associated with taxa in a given phylogeny
can be obtained directly in R.

In ecology, the development of the field of community phylogenetics
\citep{Webb2002} has created a need for researchers to have access to the
evolutionary relationships of species making up communities. The relative
contributions of the role of the environment (e.g., habitat filtering), and of
biotic interactions (e.g., competitive exclusion), are inferred from the
distribution of taxa on a phylogeny composed from species occurring at larger
spatial scale. R packages and other software have been developed to generate
phylogenies from species lists using taxonomic information or DNA
sequences. These phylogenies can then be used for community phylogenetics
analyses (e.g., \citep{Pearse2013}, \citep{Kembel}, \citep{Webb}), but as OTL
becomes more comprehensive, it could become a valuable resource for ecologists.

These packages, combined with the language's support for literate programming
(\citealt{Knuth1984}; \citealt{Xie2014}), make R an ideal platform for
reproducible research in phylogenetics and comparative biology, as they allow a
complete record of the steps taken in gathering, processing, and analyzing a
given data set to be produced.

Here we present \texttt{rotl}, and R package that allows users to download
phylogenetic and taxonomic data from the OTL directly in R. \texttt{rotl} takes
advantage of OTL's Application Programming Interfaces (APIs) to access sub-trees
from the synthetic Open Tree, as well as the published source trees that
contribute to the synthesis. By providing direct access to high quality
phylogenetic data in R, \texttt{rotl} fills a key gap in typical comparative
analysis workflows, and extends the degree to which R supports reproducible
research in phylogenetics and comparative biology.

\section{API services provided by OTL}

The OTL project provides four resources that serve data to users through the
APIs:

\begin{enumerate}
\item The \emph{taxonomy} used as the backbone of the tree, the Open Tree
  Taxonomy (OTT);
\item The \emph{studies} and their associated trees, some of which are chosen by
  curators to assemble the synthetic tree;
\item A \emph{taxonomic name resolution service} (TNRS) used to match taxon
  names to the Open Tree Taxonomy identifiers;
\item The \emph{synthetic tree} itself, the `Open Tree'.
\end{enumerate}

Additionally, the project provides access to other data, the Graph of Life, that
stores information about the underlying tree alignment graph \citep{Smith2013}
used to synthesize the various sources of information from which the Open Tree
is assembled. Data from this API are not intended for normal users, and its
use will not be demonstrated in this paper, although \texttt{rotl} has functions to
retrieve data it provides.

\texttt{rotl} gives users access to all endpoints provided by version 2 of the
OTL APIs. Phylogenetic trees served by the API can be imported directly into
R's memory and are represented using the \texttt{ape} \citep{Paradis2004} tree
structure (objects of class \texttt{phylo}), or can be written to files in the
Newick, NEXUS \citep{Maddison1997}, or NeXML \citep{Vos2012} file formats. This
allows researchers to use these trees either directly with other R packages, or
to be imported in other programs that make use of phylogenetic tree files.

The full Open Tree currently does not have any branch lengths associated with
it, therefore parametric comparative methods cannot be used directly on the
subtrees returned by OTL (although the OTL treestore contains the raw
published source phylogenies, complete with branch lengths and node
annotations; see below). However, resources and methods are being developed to add
branch lengths to these topological subtrees \citep[e.g.,][]{Ksepka2015} or use
topological trees to identify phylogenetically equivalent species to increase
overlap between chronograms and species trait data \citep{Pennell2015}. Without
branch lengths, these subtrees are nonetheless useful to illustrate relationships
among species, or to map traits on a phylogeny.

%% any examples available?

% Tree versioning? Replicability of analyses

\section{Technical information about \texttt{rotl}}

Phylogenetic information retrieved from OTL is converted into \texttt{phylo}
objects by \texttt{rotl} using the NEXUS Class Library \citep[NCL,][]{Lewis2003}
as implemented in the \texttt{rncl} package
(\url{https://cran.r-project.org/package=rncl}). Using NCL provides robust and
efficient parsing of large trees that may contain singleton nodes labeled with
taxonomic information (i.e., a monotypic taxon). Singleton nodes are collapsed
after the tree has been parsed, making the resulting \texttt{phylo} object
compatible with all functions from the \texttt{ape} package.

The package is well-documented, and includes three package
vignettes (documents that demonstrate the use of the package and contain executable
R code). There is also an extensive test-suite that covers both the internal
functions that \texttt{rotl} uses to connect to OTL and public functions that users
apply to access and process data.


\section{Demonstrations}
\label{sec:demonstrations}

\subsection{Getting relationships from a list of taxa}
\label{sec:get-relationships}

Before a researcher can use the Open Tree to retrieve relationships among a set
of taxa, they first need to match the taxon names in their dataset with records
in the Open Tree Taxonomy (OTT). The TNRS allows users to search for taxon
names, and retreive identifiers for each matching taxon. We demonstrate the use
of the TNRS within \texttt{rotl} by searching for taxonomic records associated
with several model organisms.

<<init, echo=FALSE, purl=FALSE>>=
library(rotl)
@
<<get_taxa, cache=TRUE, purl=FALSE>>=
taxa <- tnrs_match_names(names = c("Escherichia colli",
                                   "Chlamydomonas reinhardtii",
                                   "Drosophila melanogaster",
                                   "Arabidopsis thaliana",
                                   "Rattus norvegicus",
                                   "Mus musculus",
                                   "Cavia porcellus",
                                   "Xenopus laevis",
                                   "Saccharomyces cervisae",
                                   "Danio rerio"))
@

The function \texttt{tnrs\_match\_names} returns a data frame that lists the
Open Tree identifiers as well as other information to help users ensure that the
taxa matched are the correct ones. Here, there is no ambiguity in the taxa
matched; however, as OTT includes taxa from bacteria, plants,
and animals that are regulated by different nomenclatural codes (ICNP, ICN, and
ICZN, respectively), both OTL and \texttt{rotl} provide tools to deal with names
that may represent valid taxa in more than one code. The argument \texttt{context\_name} can be used to limit potential
matches to a taxonomic group such as ``Animals'' (see the function
\texttt{tnrs\_contexts} for a complete list of possible options). When this strategy
cannot be used (as in the present example, where the tree encompasses
multiple domains), the function \texttt{inspect} lists alternative matches for a taxon
name, and \texttt{update} replaces it in the results. An example of this
approach is provided in the vignette ``How to use \texttt{rotl}?'' that
accompanies the package.

By default, approximate matching is enabled when attempting to match taxonomic
names to their OTT identifiers. Additionally, taxonomic synonyms
are included in OTT, allowing researchers to match correct
identifiers for taxon names that might include misspellings or synonyms. These
features will facilitate the tedious data cleaning process often needed when
mapping taxon names. In the example provided, both \textit{Escherichia coli}
and \textit{Saccharomyces cerevisiae} are misspelled, but the OTL TNRS finds the
correct match for these taxa.

Now that the taxon names are matched to the Open Tree identifiers, we can pass
them to the function \texttt{tol\_induced\_subtree} to retrieve the
relationships among these taxa. In turn, the tree can be plotted directly as it
is returned as \texttt{phylo} object (Figure~\ref{fig:plot_taxa}).

<<plot_taxa, fig.height=4, cache=TRUE, purl=FALSE, fig.cap="The phylogenetic tree returned by OTL for the list of model species used as an example.">>=
tree <- tol_induced_subtree(ott_ids = taxa[["ott_id"]])
plot(tree, cex = .8, label.offset = .1, no.margin = TRUE)
@

\subsection{Getting trees from studies}
\label{sec:get-tree-study}

\texttt{rotl} can also be used to retrieve trees accompanying studies that have
been submitted through the curator interface, and identify the trees that
contribute to the synthetic tree. As of March 2016, the Open Tree of Life
project stores 7755 trees from 3399 studies (each having between 0 and 61
trees), and 477 of these trees are used to assemble the synthetic tree. These
trees constitute a useful resource to reproduce or expand on a previously
published analysis, or to explore how the elucidation of relationships within
a clade has changed through time.

Criteria that can be used to search for studies or their associated trees are
available through the output of the function
\texttt{studies\_properties}. The meaning of these properties is described at:
\url{https://github.com/OpenTreeOfLife/phylesystem-api/wiki/NexSON}. Typically,
users will want to search for studies or trees based on taxon names (or their
OTT identifiers), but other criteria such as the title of the publication can be
used. Here we demonstrate how to look for and retrieve trees for studies
focusing on the family Felidae (Figure~\ref{fig:plot_cats}).

<<get_cats, purl=FALSE, fig.cap="Tree from \\citealt{Johnson2006} obtained from OTL">>=
cat_studies <- studies_find_studies(property = "ot:focalCladeOTTTaxonName",
                                    value = "felidae")
cat_studies
@

Currently only one study focused on this family is available from OTL, and a
single tree is associated with it. We can then retrieve the study and tree
identifiers, and pass them to the function \texttt{get\_study\_tree} to have the
tree in memory:

<<get_cat_tree, purl=FALSE, results='hide'>>=
cat_tree <- get_study_tree(study_id = cat_studies[["study_ids"]][1],
                           tree_id = cat_studies[["tree_ids"]][1])
cat_tree
@

<<print_cat_tree, purl=FALSE, echo=FALSE>>=
print(cat_tree, printlen = 4)
@

<<plot_cats, purl=FALSE, fig.height=4.2, echo=FALSE, fig.cap='Phylogeny of the Felidae published in \\citealt{Johnson2006} and retrieved from OTL using \\texttt{rotl}.'>>=
plot(cat_tree, label.offset=.0001, cex=.75, no.margin=TRUE)
@

When more than one tree is available for a given study, the function
\texttt{list\_trees} returns a list containing the tree identifiers for each
study. Alternatively, the function \texttt{get\_study} returns all the trees (by
default as \texttt{phylo}) associated with a particular
study. Additional details about the study (i.e., its metadata) can be obtained
using the function \texttt{get\_study\_meta}.


\subsection{How does \texttt{rotl} fit into the R package ecosystem?}
\label{sec:how-does-rotl-fit}

In recent years, R has become an essential part of the toolbox of many
researchers in evolutionary biology. R greatly facilitates the analysis of large
datasets, and allows researchers to combine methods in novel ways because many
methods for comparative analyses are implemented, and because it is a relatively
easy to use programming language. Additionally, as more data are made available
online and accessible using APIs, several packages have been developed to
interact and download these datasets directly in R, thereby enabling direct and
reproducible analyses.  Notably, the organization rOpenSci
(\url{https://ropensci.org}) has fostered a community of researchers who develop
tools and methods to facilitate the use of open data as well broaden the
adoption of open science practices in general. For instance, the
rOpenSci-developed package \texttt{TreeBase} \citep{Boettiger2012TB} allows users
to access phylogenies stored in treeBASE (\url{www.treebase.org}).
\texttt{rotl} contributes to this initiative, and greatly extends the number of
taxa for which phylogenetic data can be retrieved within R, while allowing the
data from OTL to be combined with other sources easily.

Here we show how we can obtain a map of the occurrences for a subset of the cat
species (genus \textit{Lynx}) that were included in the phylogeny retrieved from
the Felidae study above. We extract the species names from the phylogeny, and use
them to ask for the records for these species found in GBIF
(Figure~\ref{fig:cat_map}). We include the code to reproduce this figure in the
Appendix.

In addition of an introduction on how to use the package, \texttt{rotl} includes
two vignettes that demonstrate how to integrate a phylogeny and data associated
with the taxa it represents. Specifically, the ``Data mashups'' vignette
provides an example of how to retrieve a phylogeny for species a researcher may
have data for, and visualize both the phylogeny and data associated with the
species at the tips. The other vignette titled ``Meta-analysis'' demonstrates how a complete
comparative method analysis, including the gathering of data and a phylogeny, can be
performed in a single R session. We reproduce a published meta-analysis testing
for differential investment in male and female offsprings among 51 species of
birds. As new versions of the OTL API and \texttt{rotl} are released,
these vignettes will be kept up-to-date.



<<cat_map, purl=TRUE, echo=FALSE, size='small', fig.height=4.2, results='hide', message=FALSE, cache=TRUE, fig.cap='GBIF records for the species in \\textit{Lynx} included in the phylogeny associated with the study by \\citealt{Johnson2006}.'>>=
## load required packages. Versions used for the manuscript listed below.
library(rotl)     # v0.5.0
library(rgbif)    # v0.9.2
library(ggplot2)  # v2.1.0
library(maps)     # v3.1.0

## Import the felid tree using study and tree IDs discovered with
## studies_find_studies() in the manuscript
cat_tree <- get_study_tree(study_id ="pg_1981",
                           tree_id = "tree4052")

## Find the species of Lynx in the phylogeny
cat_species <- cat_tree$tip.label
lynx_species <- grep("^Lynx", cat_tree$tip.label, value = TRUE)

## Match the Lynx species to the GBIF identifiers
gbif_keys <- sapply(lynx_species,
                    function(x) name_backbone(name = x)$speciesKey,
                    USE.NAMES = FALSE)

## Search for the GBIF records for these species
lynx_loc <- occ_search(taxonKey = gbif_keys, limit = 500,
                       return = "data", fields = "minimal",
                       hasCoordinate = TRUE)

## Make a data frame of the results
lynx_loc <- do.call("rbind", lynx_loc)
names(lynx_loc)[1] <- "Species"

## Clean up the data with missing locality data
lynx_loc[["decimalLatitude"]] <- as.numeric(lynx_loc[["decimalLatitude"]])
lynx_loc[["decimalLongitude"]] <- as.numeric(lynx_loc[["decimalLongitude"]])
lynx_loc[lynx_loc[["decimalLatitude"]] == 0 &
         lynx_loc[["decimalLongitude"]] == 0,
         c("decimalLatitude","decimalLongitude")] <- c(NA, NA)
lynx_loc <- lynx_loc[complete.cases(lynx_loc), ]

## Draw the map
world <- map_data("world")

ggplot(lynx_loc) +
    annotation_map(world, fill="gray40", color="gray40") +
    geom_point(aes(y = decimalLatitude, x = decimalLongitude, color = Species),
               size = 1) +
    coord_map(projection = "mercator", orientation = c(90, 0, 0)) +
    xlab("Longitude") + ylab("Latitude") +
    theme(legend.position="top", legend.key = element_rect(fill = "gray40")) +
    ylim(c(0,72))
@



\section{Concluding remarks}
\label{sec:conclusion}

The recognition of the importance of phylogenies to account for the statistical
non-independence of species in comparative methods, and the recent development
of methods to explore trait evolution or changes in diversification rates, have
driven the need for accurate phylogenies. However, there is often a discrepancy
between taxa targeted by comparative methods, and taxa for which phylogenies are
available. We believe that by providing an easy-to-use interface to obtain
phylogenies for an arbitrary set of taxa directly in R, \texttt{rotl} will be
useful in a wide variety of contexts.

The accuracy and usefulness of the data provided by OTL relies on the community
to make generated phylogenies (and their metadata) digitally available as tree
files (i.e., Newick, NEXUS or NeXML). We strongly encourage researchers to
submit their published phylogenies to OTL using the curator interface
(\url{https://tree.opentreeoflife.org/curator}). By facilitating the discovery and
re-use of published trees and of the synthetic Open Tree, we hope \texttt{rotl} will
contribute to the wider adoption of best practices to make phylogenetic
information available and re-usable.


\section{Availability}
\label{sec:availability}

\texttt{rotl} is free, open source, and released under a Simplified BSD
license. Stable versions are available from the CRAN repository
(\url{https://cran.r-project.org/package=rotl}), and development versions are
available from GitHub (\url{https://github.com/ropensci/rotl}). This manuscript
was built using version 0.4.1.999-rotl-ms.1
(\url{https://github.com/ropensci/rotl/tree/v0.4.1.999-rotl-ms.1}). The package
is under active development, and authors welcome bug reports or feature requests
via the GitHub repository.  The source for this manuscript is available on
GitHub (\url{https://github.com/fmichonneau/rotl-ms}).

Python (\url{https://github.com/OpenTreeOfLife/pyopentree}) and Ruby
(\url{https://github.com/SpeciesFileGroup/bark}) libraries to interact
with the OTL APIs are also available.


\section{Acknowledgments}
\label{sec:acknowledgments}

We would like to thank the organizers of the OpenTree of Life APIs hackathon
that was held at the University of Michigan, Ann Arbor, 15-19 September, 2014,
where the development of \texttt{rotl} was started. We would also like to thank
Scott Chamberlain (rOpenSci) for providing a thorough code review and Shinichi
Nakagawa and Alistair Senior for their help in developing the package's
meta-analysis vignette. DJW was supported by NIH Grant R01-GM101352. FM was
supported by iDigBio, and therefore this material is based upon work supported
by the National Science Foundationâ€™s Advancing Digitization of Biodiversity
Collections Program (Cooperative Agreement EF-1115210).

\bibliographystyle{sysbio}

\bibliography{rotl-manuscript}

\end{document}
